packages/xrpl/test/connection.test.ts

```diff
@@ -0,0 +1,990 @@
+          assert.deepEqual(data, {
+            error: 'slowDown',
+            error_message: 'slow down',
+          })
+          resolve()
+        })
+      })
+
+      // @ts-expect-error -- Testing private member
+      clientContext.client.connection.onMessage(
+        JSON.stringify({
+          error: 'slowDown',
+          error_message: 'slow down',
+        }),
+      )
+
+      await errorPromise
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'propagates RippledError data',
+    async () => {
+      const request = { command: 'subscribe', streams: 'validations' }
+      clientContext.mockRippled?.addResponse(
+        request.command,
+        rippled.subscribe.error,
+      )
+
+      await clientContext.client.request(request).catch((error) => {
+        assert.strictEqual(error.name, 'RippledError')
+        assert.strictEqual(error.data.error, 'invalidParams')
+        assert.strictEqual(error.message, 'Invalid parameters.')
+        assert.strictEqual(error.data.error_code, 31)
+        assert.strictEqual(error.data.error_message, 'Invalid parameters.')
+        assert.deepEqual(error.data.request, {
+          command: 'subscribe',
+          id: 0,
+          streams: 'validations',
+        })
+      })
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'unrecognized message type',
+    async () => {
+      const unknownPromise = new Promise<void>((resolve) => {
+        // This enables us to automatically support any
+        // new messages added by rippled in the future.
+        clientContext.client.connection.on('unknown', (event) => {
+          assert.deepEqual(event, { type: 'unknown' })
+          resolve()
+        })
+      })
+
+      // @ts-expect-error -- Testing private member
+      clientContext.client.connection.onMessage(
+        JSON.stringify({ type: 'unknown' }),
+      )
+
+      await unknownPromise
+    },
+    TIMEOUT,
+  )
+
+  // it('should clean up websocket connection if error after websocket is opened', async function () {
+  //   await clientContext.client.disconnect()
+  //   // fail on connection
+  //   // @ts-expect-error -- Testing private members
+  //   clientContext.client.connection.subscribeToLedger =
+  //     async (): Promise<void> => {
+  //       throw new Error('error on _subscribeToLedger')
+  //     }
+  //   try {
+  //     await clientContext.client.connect()
+  //     throw new Error('expected connect() to reject, but it resolved')
+  //   } catch (err) {
+  //     assert(err.message === 'error on _subscribeToLedger')
+
+  //     // _ws.close event listener should have cleaned up the socket when disconnect _ws.close is run on connection error
+  //     // do not fail on connection anymore
+  //     // @ts-expect-error -- Testing private members
+  //     clientContext.client.connection.subscribeToLedger =
+  //       async (): Promise<void> => {
+  //         // Ignore this function
+  //       }
+  //     await clientContext.client.connection.reconnect()
+  //   }
+  // })
```

Comment:
Why is this test commented out?

Reply:
Please add a followup. note for dealing with the commented-out tests.

Reply:
Done